### DISCUSSION TITLE
Ability to add custom checkout fields

### DISCUSSION THREAD

**Discussion by pohlstrategic** [woocommerce/woocommerce#45080] [WooCommerce, Blocks, and Site Editing]
## What do you like about the Cart & Checkout blocks?

Love the functionality of the layout and simplicity of implementation.

## What do you think is missing from the Cart & Checkout blocks?

Ability to add custom checkout fields. Do you have any timeline for this feature, ability to integrate with an existing plugin or simple way to add a hook / custom code?

**Comment by senadir** [comment]
Hey, we're actively working on this, either via support from Checkout Field Editor, or via an integration point. We don't have a specific timeline for this because there're a lot of things that are still unknown, but expect more updates during the months as we tackle this issue.

For the time being, I'm going to close this issue because there's nothing actionable we can do on it. Thank you for your feedback.

**Reply by PlatonB** [reply]
WooCommerce has the ability to sell downloadable products. But there is no way to remove the address field. This is very strange. It looks like a secret collaboration with Checkout Field Editor developer;).

**Reply by senadir** [reply]
We're the developers of Checkout Field Editor 😄

**Reply by PlatonB** [reply]
It is mentioned @Automattic in your profile. Hopefully, the partnership and shared employees between ThemeHigh and Automattic doesn't lead to a conflict of interest. I'm really waiting for at least the basic ability to customize checkout fields in the WooCommerce core.

**Reply by MonteLogic** [reply]
@PlatonB I am working on making my own solution but the lack of documentation is certainly an ineptitude of this project. You pretty much have to be an **expert** in state management with the use of Context and State Selectors. Also there's few comments in the files so you pretty much have to figure it all out yourself. So, with these hurdles maybe conspiracy theories aren't that far fetched. 

But on a positive note, at least they have SOMETHING that uses Gutenberg. But this project -- at this present moment -- is definitely not of the quality we've come to expect from @Auttomatic. I do look forward to the point in which the project will meet the threshold for quality expected. 

And yes if you can't tell I'm very upset about having to figure this stuff out with only a few pages of documentation.

**Reply by senadir** [reply]
@MonteLogic Checkout field editor is still on our roadmap and we plan to include that functionality directly into Checkout block.

I understand your frustration toward lack of documentation, we're trying to ramp up our documentation status once we reach more stable phases of extensibility API, and we're ramping up more developer relations engineers who would help in that effort.

Building a whole Checkout from scratch, while taking into consideration the back history WooCommerce has isn't an easy task as you.

**Reply by MonteLogic** [reply]
@senadir This [discussion](https://github.com/woocommerce/woocommerce-blocks/discussions/5483) mentions the same functionality we are discussing and is over a year old.  

Also, this is @Auttomatic team we're talking about the people who started Gutenberg so I would think that their ability to iterate on their own projects with their OWN tech would be second to none.

**Reply by MonteLogic** [reply]
With this being said I DO appreciate the efforts @Auttomatic is making but I think the push to Gutenberg has a lot of Devs in the WordPress space skeptical of the Gutenberg roadmap therefore when the team whose heading the project isn't making progress in a way we expect, it can validate some of their arguments viz. the skeptic's arguments. I don't want that, I want Gutenberg to flourish.

**Reply by MonteLogic** [reply]
Pardon me if I have said something that has come across as rude.

**Comment by masteradhoc** [comment]
+1

**Comment by Stella3** [comment]
Any update on this? It seems like such a basic feature. More and more I'm being pulled away from WooCommerce.

**Comment by thewebscaper** [comment]
Also here to ask for an update on this. I am switching over to the block checkout on all my client sites because it just is so much nicer - but half of these clients are using plugins to collect essential information at the checkout about the order, and none of these are able to be shown in the block checkout. 
The ability to collect additional information at checkout is a must - my understanding is there are no hooks provided to plugin developers to be able to display their custom fields. Either build it into the block - or open it up to add on by plugin.

**Comment by ant-topps** [comment]
We also recently moved over the Cart and Checkout Blocks. It aligns better with the one-page checkout experience users expect. Obviously, we hope to see increased conversion rates as well. But despite the "new" blocks being around for a number of years, the support is seriously lacking, even Woo themselves haven't updated their own paid plugins.... 

We've turned off a few payment gateways that didn't support the block experience. Some eventually updated their plugins, which is great. 

But to the point at hand, we operate a B2B wholesale business and customers need to capture their VAT numbers, PO numbers and Company Registration Numbers (occasionally) to the order ie via checkout billing fields. 

This could be stored against the user profile so that it can be;

1. used future orders,
2. email confirmations,
3. API to ERP platforms,
4. VAT validation (maybe),
5. updated via the manage account section,
6. etc.

**Comment by senadir** [comment]
custom fields (we call them additional fields) are all possible https://developer.woocommerce.com/docs/cart-and-checkout-additional-checkout-fields/
